### PR TITLE
chore(ci): improve php versions

### PR DESCRIPTION
**Pull request by williancs**
Improve `mongolid-laravel` to support PHP Versions 7.3, 7.4, 8.0 and 8.1

Changes that have been done

- CI
- Code Standard
- Dev dependencies

### CODE REVIEW

**Review by ezandonai — APPROVED** [review]

File: src/LegacyMongolidModel.php [hunk 1]
```diff
@@ -26,38 +27,38 @@
 abstract class LegacyMongolidModel extends LegacyRecord
 {
     /**
-     * Validation rules.
+     * Public local mock.
```
[the remark this answers is not in the document]
Comment (williancs) [reply src/LegacyMongolidModel.php:30]: We are't messing with the code at this level, just making it compatible with versions equal or greater than php 7.3
These changes will be made in future PRs

File: docker-compose.yml [hunk 2]
```diff
@@ -10,7 +10,13 @@ services:
...
   db:
     image: mongo:4.2
-    command: mongod
+    command: mongod --wiredTigerCacheSizeGB 0.25
+    deploy:
+      resources:
+        limits:
+          memory: 1g
+        reservations:
+          memory: 512M
```
[the remark this answers is not in the document]
Comment (williancs) [reply docker-compose.yml]: Done

File: docker-compose.override.yml.example [hunk 3]
```diff
@@ -1 +1,11 @@
...
+
+services:
+  db:
+    command: mongod --wiredTigerCacheSizeGB 0.25
+    deploy:
+      resources:
+        limits:
+          memory: 1g
+        reservations:
+          memory: 512M
```
Comment (diegofelix) [inline docker-compose.override.yml.example:11]: line feed here

**Review by diegofelix — APPROVED** [review]

**Review by edipoReboucas — APPROVED** [review]
Parece genuíno.

**Pull request merged**